File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int,int>>q;
        unordered_map<int,int>mp;
        vector<int>ans;
        for(int x:nums){
        
            mp[x]++;
        }
        
        for(auto &p : mp){
            q.push({p.second,p.first});    //{freq ,value} aise krenge  kyuki pq pehle column ko sort krti hai
            // and apne ko freq sort chahiye isiliye p.sec ko pehle likha
        }

        for(int i=0;i<k;i++){
            ans.push_back(q.top().second);
            q.pop();
        }
        return ans;
    }
};

File: Data Structures & Algorithms/assign-cookies/submission-1.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        ans=0
        g.sort()
        s.sort()

        i=0
        j=0

        while i<len(g) and j<len(s):
            if g[i]<=s[j]:
                ans+=1
                i+=1
                j+=1
            else:
                j+=1    

        
        return ans        